algosdk/testing/teal_blackbox.py

```diff
@@ -0,0 +1,995 @@
+from base64 import b64decode
+import csv
+from dataclasses import dataclass
+from enum import Enum
+import io
+from inspect import signature
+from tabulate import tabulate
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
+
+from algosdk.v2client.algod import AlgodClient
+from algosdk.testing.dryrun import (
+    ZERO_ADDRESS,
+    assert_error,
+    assert_no_error,
+    Helper as DryRunHelper,
+)
+
+ExecutionMode = Enum("ExecutionMode", "Signature Application")
+
+DryRunProperty = Enum(
```

Comment:
@tzaffi Beyond avoiding the declaration of enum indices, what other advantages does the _functional API_ (https://docs.python.org/3/library/enum.html#functional-api) provide for enum declaration?

* I also prefer the functional API's conciseness, but it seems to be mostly incompatible with at least PyCharm.  PyCharm seems unable to parse specific enumeration values from these strings.
* It's unclear if the limitation extends to mypy.  I made a minor change in pyteal where I referred to a non-existent enumeration value.  mypy did _not_ report an error.  https://mypy.readthedocs.io/en/stable/literal_types.html#enums suggests mypy does work with other forms of enum definitions.
* The IDE limitation feels hampering.  I think it'd make refactoring (e.g. finding usages) quite difficult.
* I'm wondering if we can get most of the benefit in a way that's more compatible with the tooling?
  * For example, https://docs.python.org/3/library/enum.html#using-automatic-values offers a middle ground on the manual index definition concern.

Reply:
👍 - ok, I'll go with `auto()`

Reply:
@tzaffi Thanks for bringing in `Enum` with `auto()` via https://github.com/algorand/py-algorand-sdk/commit/680e432bf586f59605f32e243690277bff715ee0.  It's working well with IDE facilities.